indico/util/placeholders.py

```diff
@@ -55,13 +55,17 @@ def get_regex(cls, **kwargs):
         return re.compile(r'\{%s}' % re.escape(cls.name))
 
     @classmethod
-    def replace(cls, text, **kwargs):
+    def replace(cls, text, escape_html=True, **kwargs):
         """Replaces all occurrences of the placeholder in a string
 
         :param text: The text to replace placeholders in
+        :param escape_html: whether HTML escaping should be done
         :param kwargs: arguments specific to the placeholder's context
         """
-        return cls.get_regex(**kwargs).sub(escape(cls.render(**kwargs)), text)
+        rendered_text = cls.render(**kwargs)
```

Comment:
Kind of unrelated, but it would be nice if we only called `render()` when the regex actually matches something to be replaced. Otherwise we always call `render()` on every single placeholder even if it's never used.

Untested:

```python
def replace(cls, text, escape_html=True, **kwargs):
    rendered = []
    
    def _replace(m):
        if rendered:
            return rendered[0]
        rendered.append(cls.render(**kwargs))
        if escape_html:
            rendered[0] = escape(rendered[0])
        return rendered[0]

    return cls.get_regex(**kwargs).sub(_replace, text)
```